common/models/report.js

```diff
@@ -1,110 +1,43 @@
+var countries = require('country-data').countries;
 module.exports = function(Report) {
-  Report.listReports = function(key, callback) {
-    var allowed_keys = ["test_name", "probe_cc", "probe_asn", "start_time"];
-    if (allowed_keys.indexOf(key) === -1) {
-      callback(new Error("Key must be in " + allowed_keys));
-    }
-    var filter = {
-      fields: {}
-    }
-    allowed_keys.forEach(function(k){
-      filter.fields[k] = true;
-    });
-    Report.find(filter, function(err, data){
-      var reports = {};
-      data.forEach(function(item) {
-        var key_value = item[key];
-        reports[key_value] = reports[key_value] || {count: 0};
-        reports[key_value]["count"] += 1;
-      });
 
-      callback(err, reports);
-    });
-  }
- 
- /**
-  * Returns all the reports that have been collected
-  * @param {string[]} country_code country codes to filter by
-  * @param {number} limit maximum number of results to return
-  * @callback {Function} callback Callback function
-  * @param {Error|string} err Error object
-  * @param {Object} result Result object
-  */
-  Report.findReports = function(country_code, input, fields, limit, callback) {
-    var filter = {
-      limit: limit,
-      fields: {},
-      where: {}
-    };
-    if (country_code && country_code.length) {
-      filter.where["probe_cc"] = {inq: country_code};
+  Report.countByCountry = function(alpha2, callback) {
+    var ds = Report.dataSource;
+    var sql = "SELECT probe_cc, count FROM country_counts_view";
+
+    // This is clearly not ideal, does Strongloop provide for sanitization?
+    // Surely their query language is a bit stronger than this?
+    if (alpha2 !== undefined && alpha2 !== null) {
```

Comment:
@hellais What are your thoughts on this? It doesn't seem very secure, but I wasn't sure whether Loopback does sanitization here.